src/testnet/burnchain.rs

```diff
@@ -0,0 +1,141 @@
+use std::sync::mpsc;
+use std::thread;
+use std::time;
+use std::sync::{Arc, Mutex};
+
+use super::{Config};
+
+use burnchains::{Burnchain, BurnchainBlockHeader, BurnchainHeaderHash, BurnchainBlock, Txid};
+use burnchains::bitcoin::BitcoinBlock;
+use chainstate::burn::db::burndb::{BurnDB};
+use chainstate::burn::{BlockSnapshot};
+use chainstate::burn::operations::{BlockstackOperationType};
+use util::hash::Sha256Sum;
+
+/// BurnchainSimulator is simulating a simplistic burnchain.
+/// 
+/// When calling start(), a thread is being detached, looping, sleeping and firing some events, and a tuple (tx, rx) 
+/// is being returned so that the caller can be receiving new blocks, and transmitting ops in a simulated mem_pool.
+pub struct BurnchainSimulator {
+    mem_pool: Arc<Mutex<Vec<BlockstackOperationType>>>,
+    db: Option<Arc<Mutex<BurnDB>>>,
+}
+
+impl BurnchainSimulator {
+
+    pub fn new() -> Self {
+        Self {
+            mem_pool: Arc::new(Mutex::new(vec![])),
+            db: None,
+        }
+    }
+    
+    pub fn start(&mut self, config: &Config) -> (mpsc::Receiver<(BlockSnapshot, Vec<BlockstackOperationType>, Arc<Mutex<BurnDB>>)>, mpsc::Sender<BlockstackOperationType>) {
```

Comment:
Not a bug, but I'd recommend creating a dedicated `struct` for containing the state in `(BlockSnapshot, Vec<BlockstackOperationType>, Arc<Mutex<BurnDB>>)`, instead of returning a long tuple.

Reply:
Addressed with https://github.com/blockstack/blockstack-core/pull/1202/commits/4ec6caf273aec0bf70ca5070cfad7e7a3fd0b0cf